src/core/markovProcessGraph.js

```diff
@@ -0,0 +1,579 @@
+          SEED_IN,
+          ...NodeAddress.toParts(node.address)
+        ),
+        reversed: false,
+        src: node.address,
+        dst: SEED_ADDRESS,
+        transitionProbability: seed.alpha,
+      });
+    }
+
+    // Add minting (seed-out) edges
+    {
+      let totalNodeWeight = 0.0;
+      const positiveNodeWeights: Map<NodeAddressT, number> = new Map();
+      for (const {address, weight} of _nodes.values()) {
+        if (weight > 0) {
+          totalNodeWeight += weight;
+          positiveNodeWeights.set(address, weight);
+        }
+      }
+      if (!(totalNodeWeight > 0)) {
+        throw new Error("No outflow from seed; add cred-minting nodes");
+      }
+      for (const [address, weight] of positiveNodeWeights) {
+        addEdge({
+          address: EdgeAddress.append(
+            SEED_OUT,
+            ...NodeAddress.toParts(address)
+          ),
+          reversed: false,
+          src: SEED_ADDRESS,
+          dst: address,
+          transitionProbability: weight / totalNodeWeight,
+        });
+      }
+    }
+
+    /**
+     * Find an epoch node, or just the original node if it's not a
+     * scoring address.
+     */
+    const rewriteEpochNode = (
+      address: NodeAddressT,
+      edgeTimestampMs: number
+    ): NodeAddressT => {
+      if (!_scoringAddresses.has(address)) {
+        return address;
+      }
+      const epochEndIndex = sortedIndex(timeBoundaries, edgeTimestampMs);
+      const epochStartIndex = epochEndIndex - 1;
+      const epochTimestampMs = timeBoundaries[epochStartIndex];
+      return epochNodeAddressToRaw({
+        type: "EPOCH_NODE",
+        owner: address,
+        epochStart: epochTimestampMs,
+      });
+    };
+
+    // Add graph edges. First, split by direction.
+    type _UnidirectionalGraphEdge = {|
+      +address: EdgeAddressT,
+      +reversed: boolean,
+      +src: NodeAddressT,
+      +dst: NodeAddressT,
+      +timestamp: TimestampMs,
+      +weight: number,
+    |};
+    const unidirectionalGraphEdges = function* (): Iterator<_UnidirectionalGraphEdge> {
+      const ewe = edgeWeightEvaluator(wg.weights);
+      for (const edge of (function* () {
+        for (const edge of wg.graph.edges({showDangling: false})) {
+          const weight = ewe(edge.address);
+          yield {
+            address: edge.address,
+            reversed: false,
+            src: edge.src,
+            dst: edge.dst,
+            timestamp: edge.timestampMs,
+            weight: weight.forwards,
+          };
+          yield {
+            address: edge.address,
+            reversed: true,
+            src: edge.dst,
+            dst: edge.src,
+            timestamp: edge.timestampMs,
+            weight: weight.backwards,
+          };
+        }
+      })()) {
+        if (edge.weight > 0) {
+          yield edge;
+        }
+      }
+    };
+
+    const srcNodes: Map<
+      NodeAddressT /* domain: (nodes with out-edges) U (epoch nodes) */,
```

Comment:
I originally interpreted this comment to mean "every node with out-edges and every epoch node will be included in this map". However:
- nodes with only zero-weight out edges are not included
- epoch nodes are only included if there was an out-edge from the corresponding scoring node with a timestamp in that epoch node's interval

suggest updating the comment for clarity

Reply:
in retrospect, this seems a little pedantic, but still correct, so maybe worth following up on.